lib/shopping_cart.rb

```diff
@@ -0,0 +1,53 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    {name: @name, capacity: @capacity}
+  end
+
+  def total_number_of_products
```

Comment:
This is really good use of `#reduce` here. If you are simply adding integers together though the `#sum` enumerable would be even better here